torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -138,7 +138,9 @@ struct GraphFuser {
     // We don't want to bother with cross-block node movements, as they
     // are not necessarily correct.
     if (node->owningBlock() != block_) return false;
-    return node->kind() == prim::FusionGroup || isSimpleMap(node);
+    return node->kind() == prim::FusionGroup
+      || node->kind() == prim::AutodiffGradSumToSize
```

Comment:
Um this is untrue. You can’t always fuse those nodes.

Reply:
Fair enough, even if the logic now is a lot more complicated than I would like (especially given that we expect to be able to fuse them again).